samples/bluetooth/peripheral_ots/src/main.c

```diff
@@ -192,13 +193,24 @@ static void ots_obj_name_written(struct bt_ots *ots, struct bt_conn *conn,
 		id_str, cur_name, new_name);
 }
 
+uint32_t ots_obj_cal_checksum(uint64_t id, off_t offset, size_t len)
+{
+	uint32_t crc;
+	uint32_t obj_index = ((id - BT_OTS_OBJ_ID_MIN) % ARRAY_SIZE(objects));
+
+	crc = crc32_ieee((const uint8_t *)(&objects[obj_index].data + offset), len);
+	printk("offset %lu len %zu crc 0x%08x\n", offset, len, crc);
+	return crc;
+}
```

Comment:
I assume this is calculating the checksum with accordance to the OTS spec. 

Suggest to add a new function, ala `bt_ots_cal_checksum` that can be used so that the application doesn't need to implement it itself

Reply:
I can't make my mind about where to implement the **"actual"** checksum calculation.

1. Let application calculate.
2. Application return address of data, let OTS service to calculate.

Since object data is opaque, OTS service can't access object data directly.

Reply:
Yeah, so basically we have

1: The application implements and calculates the checksum. However, since it needs to be return in this callback, we will be blocking whatever thread need that needs to return this value. IIRC an object can be up to 2^32-1 octets in size (roughly 4GB), and calculating the checksum of such an object could potentially take time, so calculating it in a callback isn't perhaps the best solution.

2: Application returns the pointer to the object's data, and we can implement the calculation and handling of calculating this in a different thread than the BT RX thread (e.g. the BT Long Workqueue), and send the indication when done

3: Like 1:, but where we expose a `bt_ots_calculate_checksum` function to the application, so that it doesn't need to worry about using `crc32_ieee` correctly. 

I think 2: would give the best result, as we can best control the calculation there, and it follows the same principles of the read functionality.